### PR TITLE
Add rule S6774 (`prop-types`): React components should validate prop types

### LINKED ISSUE #4142: Create rule for `react/prop-types​`: Disallow missing props validation in a React component definition
- [Reference](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/prop-types.md)
- [Implementation](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/lib/rules/prop-types.js)
- [Tests](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/tests/lib/rules/prop-types.js)

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #4142

### CODE REVIEW

File: sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S6774.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "impacts": {
+      "MAINTAINABILITY": "HIGH",
+      "RELIABILITY": "MEDIUM",
+      "SECURITY": "LOW"
+    },
+    "attribute": "CONVENTIONAL"
+  },
+  "compatibleLanguages": [
+    "JAVASCRIPT"
+  ]
```
[the remark this answers is not in the document]
Comment (alexander-kamushkin-sonarsource) [reply sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S6774.json:27]: I am not sure - the tests include TypeScript: https://github.com/jsx-eslint/eslint-plugin-react/blob/ecadb92609998520be80d64c0bd6bc5e05934aa9/tests/lib/rules/prop-types.js#L1286

On the other hand, it is indeed looks redundant for TypeScript

[the remark this answers is not in the document]
Comment (alexander-kamushkin-sonarsource) [reply sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S6774.json:27]: Maybe we should try to enable it and see how much noise it makes?

**Review by alexander-kamushkin-sonarsource — APPROVED** [review]
I am undecided if this should be enabled for typescript or not, so up to you.

File: sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S6774.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (yassin-kammoun-sonarsource) [reply sonar-plugin/javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S6774.json:27]: Tried and noisy, too noisy. Let's keep the rule enabled for JavaScript only.

**Pull request merged**